Notes about entire cluster shutdown

qhartman opened the pull request:
Hoping to save people some pain...

Signed-off-by: Quentin Hartman <qhartman@gmail.com>

gregsfortytwo commented:
We need you to sign off on commits before we can take them into the tree (this is the indication that they're appropriately licensed, etc). See https://github.com/ceph/ceph/blob/master/SubmittingPatches for details and if you need help changing the commit message. :)

jcsp commented on doc/rados/operations/operating.rst:
```diff
@@ -352,7 +352,26 @@ To stop a Ceph daemon on another node, use the following syntax::
+
+Then shutdown all the services in your cluster in the manner appropriate for your platform.
 
+.. note:: Ceph is not really intended to be entirely shutdown, and it's behaviors in this
```
>Ceph is not really intended to be entirely shutdown

I would tone town this statement a little bit -- Ceph *is* intended to function properly when entirely shut down, it just isn't quite as friendly as it could be.

qhartman commented:
@jcsp That's fair. Toned down.

liewegas commented on doc/rados/operations/operating.rst:
```diff
@@ -352,7 +352,26 @@ To stop a Ceph daemon on another node, use the following syntax::
+	It's behaviors in this case are not well validated at this point. Nothing should
+	go really wrong, but there may be some irritations associated with this operation.
+	After starting back up, if anything seems amiss, restart the daemons that are
+	misbehaving and their behavior should return to normal.
```
i'm curious what those irritations are?  i can't think of anything that should work here...

qhartman replied:
See https://www.mail-archive.com/ceph-users@lists.ceph.com/msg18514.html for details on my experience and the comments from Gregory Farnum

dachary commented:
@qhartman feel free to re-open when you have time to address the comments.

qhartman commented:
@dachary I thought I had addressed all the comments. The last one from @liewegas seemed to be asking for context, which I provided, rather than an additional change to the doc. If that's not the case, please clarify and I will address as needed.

dachary commented:
@qhartman I think sage meant that the commit should explain what the irritation is. The comment from this pull request won't show in the documentation and won't be helpful to the reader.

dachary commented:
@qhartman could you squash the four commits into one and make sure the Signed-off-by: looks like the others found in the log (single line, &lt;> around the mail). I'll let an english speaking person decide for the merge.

qhartman commented:
OK, I believe I have the commits correctly squashed and the sign-off message correctly formatted.

xiaoxichen commented on doc/rados/operations/operating.rst:
```diff
@@ -352,7 +352,29 @@ To stop a Ceph daemon on another node, use the following syntax::
+	ceph osd set noout
+
+Then shutdown all the services in your cluster in the manner appropriate for your platform.
+
```
in this context, maybe we could use "service ceph -a stop osd" ?

qhartman replied:
I didn't include that specificity intentionally. The command you reference would work on an Ubuntu machine, but not other platforms. You can see the various platform-specific incantations at the following links:

http://ceph.com/docs/master/rados/operations/operating/#id1
http://ceph.com/docs/master/rados/operations/operating/#starting-all-daemons
http://ceph.com/docs/master/rados/operations/operating/#id7

My assumption is that if a user is reading this part of the documentation they would have already become familiar with the commands to start and stop the individual services in their deployment, or would be able to readily find them. Omitting all the possible variations keeps this section concise, and prevents the need for future updates if/when things change.

xiaoxichen replied:
Sorry I am not an expert on this, but it really confuse me, in some of the doc, we explicitly use "service ceph -a stop osd" , but here you want to prevent the specificity. So what's the general rules we want to maintain?

xiaoxichen commented on doc/rados/operations/operating.rst:
```diff
@@ -352,7 +352,29 @@ To stop a Ceph daemon on another node, use the following syntax::
 	sudo service ceph -a stop osd.0
```
see here

dachary commented:
The noout is a good advice. However, the chapter in between is too general and does not belong here.

theanalyst commented on doc/rados/operations/operating.rst:
```diff
@@ -352,7 +352,29 @@ To stop a Ceph daemon on another node, use the following syntax::
+Then shutdown all the services in your cluster in the manner appropriate for your platform.
+
+.. note:: Ceph is known to complain in small ways when coming up after a complete shutdown.
+	It's behaviors in this case are not well validated at this point. Nothing should
```
behaviors/behavior are/is

dachary commented:
@qhartman feel free to re-open when you have time to address the comments
